### PR TITLE
Data Trails: Actions redesign and overview tab

### LINKED ISSUE #79594: Data Trails: Build Overview Tab
**See Figma file for most up to date designs and more details**: https://www.figma.com/file/k87knv2idoa0EIsLpQOWkR/Data-Trails?type=design&node-id=411%3A11566&mode=design&t=gmzS9pM7p2lsLAF0-1

Quick prototype (note some UI elements have changed since this): https://github.com/grafana/grafana/assets/5264102/6780dbf5-b09b-46c9-b560-966013ff723c

There is some discussion about what the graph summary should look like for histograms and it's not super fleshed out. See discussion: https://raintank-corp.slack.com/archives/C0630AYC4SY/p1702605993411659

### PR DESCRIPTION
**Pull request by adrapereira**
**What is this feature?**

This PR moves the Breakdown, Logs and Related Metrics action buttons to tabs and all other actions to a new actions bar located on the right hand side of the tabs row. Additionally it adds a new Overview tab which is selected by default when opening a metric scene.

The `Breakdown`, `Logs` and `Related Metrics` actions already worked as tabs but were using button components to swap between them, so moving it to using a tabs component was straightforward. 

The `Add to dashboard`, `Share`, `Bookmark` and `Explore` buttons were reorganised and moved to a new actions bar to the right of the tabs. On narrow viewports the actions are placed above the tabs.

This PR also includes a new `Overview` tab with general information about the selected metrics. It displays the description, type and unit, extracted from the metrics metadata (when available). It also shows the available labels for that metric with the count of the label values next to it. Clicking a label links to the `Breakdown` tab with that label selected. 

Extra: newly created trails use the datasource from the most recently used trail, when available, or the configured default datasource if its type is Prometheus. 

**Which issue(s) does this PR fix?**:

Fixes #78180 
Fixes #79594 

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/trails/utils.ts [hunk 1]
```diff
@@ -47,6 +50,21 @@ export function getMetricSceneFor(model: SceneObject): MetricScene {
...
+  if (typeof initialDS === 'string') {
+    initialDSValue = initialDS;
+  } else {
+    const promDatasources = getDatasourceSrv().getList({ type: 'prometheus' });
+    if (config.defaultDatasource && promDatasources.find((mds) => mds.uid === config.defaultDatasource)) {
+      initialDSValue = config.defaultDatasource;
+    }
+  }
+  return initialDSValue;
+}
```
Comment (torkelo) [inline public/app/features/trails/utils.ts:66]:
```suggestion
export function getDatasourceForNewTrail(): string | undefined {
  const prevTrail = getTrailStore().recent[0];
  if (!prevTrail) {
    return config.defaultDatasource;
  }

  const prevDataSource = sceneGraph.interpolate(prevTrail.resolve(), VAR_DATASOURCE_EXPR);

  if (typeof prevDataSource === 'string' && prevDataSource.length > 0) {
    return prevDataSource;
  }

  const promDatasources = getDatasourceSrv().getList({ type: 'prometheus' });
  if (promDatasources.length > 0) {
    return promDatasources.find((mds) => mds.uid === config.defaultDatasource) ?? promDatasources[0].uid;
  }

  return undefined;
}
```

Maybe something like that makes sense? use last trail data source, if no data source there, take the default if it's prometheus, if default is not prometheus take first prometheus ds

File: public/app/features/trails/MetricOverviewScene.tsx [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+
+  private updateMetadata() {
+    const metricScene = getMetricSceneFor(this);
+    const metric = metricScene.state.metric;
+    const dsUid = sceneGraph.getAncestor(this, DataTrail).state.$variables?.getByName(VAR_DATASOURCE)?.getValue();
+    if (typeof dsUid === 'string') {
+      getDatasourceSrv()
+        .get(dsUid)
+        .then((ds) => {
+          const langProvider: PrometheusLanguageProvider = ds.languageProvider;
```
Comment (torkelo) [inline public/app/features/trails/MetricOverviewScene.tsx:100]: We should try to refactor this so we can share the same language provider here as the one fetched in updateLanguageProvider

File: public/app/features/trails/MetricOverviewScene.tsx [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+        this.state.languageProvider?.fetchSeriesValuesWithMatch(l.value, match).then((res) => {
+          this.setState({
+            labels: this.state.labels.map((label) =>
+              label.value === l.value ? { ...label, count: res.length } : label
+            ),
+          });
+        });
+      }
+    });
+  }
```
Comment (torkelo) [inline public/app/features/trails/MetricOverviewScene.tsx:125]: I think we can skip this for now, I don't want to do a N+ query here. This should be done in a single call (ie fetch labels and count in a single instant call). But that needs some changes to variable system and Prometheus variable queries

File: public/app/features/trails/MetricOverviewScene.tsx [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+
+  private getLabelOptions(variable: QueryVariable) {
```
Comment (torkelo) [inline public/app/features/trails/MetricOverviewScene.tsx:127]: this function looks identical from the one in BreakdownScene, wonder if we can extract it to a util function somehow

**Review by torkelo — COMMENTED** [review]
looks good, but afraid of the label count and the complexity / performance of that solution. I want that count to come from the variable system itself and the variable query (so it can be done in a single call)

File: public/app/features/trails/MetricScene.tsx [hunk 5]
```diff
@@ -113,57 +113,82 @@ export class MetricActionBar extends SceneObjectBase<MetricActionBarState> {
       setBookmarked(!isBookmarked);
     };
 
+    if (!actionView) {
+      metricScene.setActionView(actionViewsDefinitions[0].value);
+    }
```
Comment (torkelo) [inline public/app/features/trails/MetricScene.tsx:118]: you can't set state from react render context (will lead to react warnings)

**Review by torkelo — COMMENTED** [review]
Wanted to try some refactors to see if they actually worked (instead of proposing them here).

You can view them here: 
https://github.com/grafana/grafana/compare/andre/data-trails-redesign...andre/data-trails-redesign-tlo-refactor 

* Moves variable option subscriptions to the react part (via useState) 
* uses _variableDependency to get notified if data source changes 
* Create text links to the breakdown view instead of buttons 
* I don't think we need to subscribe to metric changes as that creates new MetricScene's

File: public/app/features/trails/ActionTabs/MetricOverviewScene.tsx [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+        {loading ? (
+          <div>Loading...</div>
+        ) : (
+          <>
+            <Stack direction="column" gap={0.5}>
+              <div className={styles.label}>Description</div>
+              {metadata?.help ? <div>{metadata?.help}</div> : <i>No description available</i>}
+            </Stack>
+            <Stack direction="column" gap={0.5}>
+              <div className={styles.label}>Type</div>
```
Comment (torkelo) [inline public/app/features/trails/ActionTabs/MetricOverviewScene.tsx:123]: use <Text weight="medium" /> here

File: public/app/features/trails/ActionTabs/MetricOverviewScene.tsx [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+
+  private _onActivate() {
+    this.updateMetadata();
+
+    const metricScene = getMetricSceneFor(this);
+    metricScene.subscribeToState((newState, oldState) => {
+      if (newState.metric !== oldState.metric) {
+        this.updateMetadata();
+      }
+    });
```
Comment (torkelo) [inline public/app/features/trails/ActionTabs/MetricOverviewScene.tsx:63]: I don't think this is needed, I could not get line 61 to trigger. When a metric is changed we build a new MetricScene (which builds a new MetricOverview),

File: public/app/features/trails/ActionTabs/LogsScene.tsx [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+import { SelectMetricAction } from '../SelectMetricAction';
+import { LOGS_METRIC } from '../shared';
+
+export function buildLogsScene() {
+  return new SceneFlexItem({
+    $data: new SceneQueryRunner({
+      queries: [
+        {
+          refId: 'A',
+          datasource: { uid: 'gdev-loki' },
```
Comment (darrenjaneczek) [inline public/app/features/trails/ActionTabs/LogsScene.tsx:12]: Is there a way to use `${datasource}` at this stage vs `gdev-loki`?

File: public/app/features/trails/ActionTabs/MetricOverviewScene.tsx [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+        )}
+      </Stack>
+    );
+  };
+}
+
+function getStyles(theme: GrafanaTheme2) {
+  return {
+    label: css({
+      fontWeight: 'bold',
```
Comment (darrenjaneczek) [inline public/app/features/trails/ActionTabs/MetricOverviewScene.tsx:135]: Should this be `'medium'` as per:
https://github.com/grafana/grafana/pull/80216/files#r1448674972

File: public/app/features/trails/MetricScene.tsx [hunk 10]
```diff
@@ -57,16 +62,17 @@ export class MetricScene extends SceneObjectBase<MetricSceneState> {
...
       }
     } else if (values.actionView === null) {
       this.setActionView(undefined);
     }
   }
 
-  public setActionView(actionViewDef?: ActionViewDefinition) {
+  public setActionView(actionView?: ActionViewType) {
     const { body } = this.state;
+    const actionViewDef = actionViewsDefinitions.find((v) => v.value === actionView);
```
Comment (darrenjaneczek) [inline public/app/features/trails/MetricScene.tsx:75]: One more of these, and we might want a `getActionView(value)`.

File: public/app/features/trails/MetricScene.tsx [hunk 11]
```diff
@@ -115,49 +119,92 @@ export class MetricActionBar extends SceneObjectBase<MetricActionBarState> {
       setBookmarked(!isBookmarked);
     };
 
+    if (!actionView) {
+      metricScene.setActionView(actionViewsDefinitions[0].value);
```
Comment (darrenjaneczek) [inline public/app/features/trails/MetricScene.tsx:123]: Do we always trust that we will want index 0, or should we just say `'overview'` here?

**Review by darrenjaneczek — APPROVED** [review]
Looks okay, and it seems you captured the primary concerns.
We can always open subsequent PR work if there is more redesign needed.

I left a few remarks for consideration, but nothing to block a merge at this time.

**Pull request merged**